File: back_igm/core/models/plan.py
from typing import Dict, Any


class Plan:
    def __init__(self, id: str, name: str, upTo: int, downTo: int, costPerProducts: int):
        self.id = id
        self.name = name
        self.costPerProducts = costPerProducts
        self.upTo = upTo
        self.downTo = downTo

    @staticmethod
    def fromJson(data: Dict[str, Any]) -> "Plan":
        return Plan(
            id=str(data["id"]),
            name=str(data["name"]),
            upTo=int(data["upTo"]),
            downTo=int(data["downTo"]),
            costPerProducts=int(data["costPerProducts"])
        )

    def toJson(self) -> Dict[str, Any]:
        return {
            "id": self.id,
            "name": self.name,
            "upTo": self.upTo,
            "downTo": self.downTo,
            "costPerProducts": self.costPerProducts
        }

File: front_igm/CustomerView/models/Plan.js

// plan free 1 to 9
// plan basic 10 to 100
// plan seller 101 to 1000
// plan shop 1001 to 100000
// plan unic * this is for own sistem for them *


export class Plan {
    constructor(id, name, upTo, downTo, costPerProducts) {
        this.id = id;                     // string
        this.name = name;                 // string
        this.costPerProducts = costPerProducts; // number
        this.upTo = upTo;                 // number
        this.downTo = downTo;             // number
    }

    static fromJson(json) {
        if (!json) return null;

        return new Plan(
            String(json.id),
            String(json.name),
            Number(json.upTo),
            Number(json.downTo),
            Number(json.costPerProducts)
        );
    }

    toJson() {
        return {
            "id": this.id,
            "name": this.name,
            "upTo": this.upTo,
            "downTo": this.downTo,
            "costPerProducts": this.costPerProducts
        };
    }
}